src/Form/input-classifiers.test.js

```diff
@@ -0,0 +1,112 @@
+    cases.forEach(({text = '', expectedResult}) => {
+        it(`test date format for ${text}`, () => {
+            const {label, input} = createElements()
+            input.autocomplete = 'cc-exp'
+            if (Math.random() < 0.5) {
+                input.placeholder = text
+            } else {
+                label.textContent = text
+            }
+
+            expect(getCCFieldSubtype(input)).toBe('expiration')
+
+            expect(getUnifiedExpiryDate(input, 12, 2025)).toBe(expectedResult)
+        })
+    })
+}
+
+afterEach(() => {
+    document.body.innerHTML = null
+})
+
+describe('test', () => {
+    it('should match the selector for cardNumber', () => {
+        const {input} = createElements()
+        input.autocomplete = 'cc-number'
+        expect(getCCFieldSubtype(input)).toBe('cardNumber')
+    })
+
+    const ccLabeltestCases = {
+        cardName: [
+            {text: 'credit card name'},
+            {text: 'name on card'},
+            {text: 'card holder'},
+            {text: 'card owner'},
+            {text: 'card number', shouldMatch: false}
+        ],
+        cardNumber: [
+            {text: 'Credit Card Number'},
+            {text: 'number on card'},
+            {text: 'card owner', shouldMatch: false}
+
+        ],
+        expirationMonth: [
+            {text: 'expiry month'},
+            {text: 'expiration month'},
+            {text: 'exp month'},
+            {text: 'card expiry mo'},
+            {text: 'Credit Card Number', shouldMatch: false},
+            {text: 'expiry year', shouldMatch: false},
+            {text: 'expiration year', shouldMatch: false},
+            {text: 'exp year', shouldMatch: false},
+            {text: 'card expiry yy', shouldMatch: false}
+        ],
+        expirationYear: [
+            {text: 'expiry year'},
+            {text: 'expiration year'},
+            {text: 'exp year'},
+            {text: 'card expiry yy'},
+            {text: 'Credit Card Number', shouldMatch: false},
+            {text: 'expiry month', shouldMatch: false},
+            {text: 'expiration month', shouldMatch: false},
+            {text: 'exp month', shouldMatch: false},
+            {text: 'card expiry mo', shouldMatch: false}
+        ]
+    }
+    testRegexForCCLabels(ccLabeltestCases)
+
+    const unifiedExpirationDateCases = [
+        {text: 'mm-yyyy', expectedResult: '12-2025'},
+        {text: 'mm/yyyy', expectedResult: '12/2025'},
+        {text: '__-____', expectedResult: '12-2025'},
+        {text: 'mm-yy', expectedResult: '12-25'},
+        {text: 'i.e. 10-2022', expectedResult: '12-2025'},
+        {text: 'MM-AAAA', expectedResult: '12-2025'},
+        {text: 'mm_jj', expectedResult: '12_25'},
+        {text: 'mm.yy', expectedResult: '12.25'},
+        {text: 'mm - yy', expectedResult: '12-25'},
+        {text: 'mm yy', expectedResult: '12 25'},
+        {text: 'ie: 08.22', expectedResult: '12.25'}
+    ]
+    testUnifiedExpirationDate(unifiedExpirationDateCases)
```

Comment:
[Jest has a built in `each`](https://jestjs.io/docs/api#testeachtablename-fn-timeout) we could use here? For readability, it might also be worth inlining the extracted test stuff? 
What do you think of the below? (I've also split it to be two tests rather than using `Math.random`)

```js
    describe("Unified Expiration Date", () => {
        describe.each([
            { text: 'mm-yyyy', expectedResult: '12-2025' },
            { text: 'mm/yyyy', expectedResult: '12/2025' },
            { text: '__-____', expectedResult: '12-2025' },
            { text: 'mm-yy', expectedResult: '12-25' },
            { text: 'i.e. 10-2022', expectedResult: '12-2025' },
            { text: 'MM-AAAA', expectedResult: '12-2025' },
            { text: 'mm_jj', expectedResult: '12_25' },
            { text: 'mm.yy', expectedResult: '12.25' },
            { text: 'mm - yy', expectedResult: '12-25' },
            { text: 'mm yy', expectedResult: '12 25' },
            { text: 'ie: 08.22', expectedResult: '12.25' }
        ])('when checking for $text', ({ text, expectedResult }) => {
            let elements

            beforeEach(() => {
              elements = createElements()
              elements.input.autocomplete = 'cc-exp'
            })

            it("matches for placeholder text", () => {
              elements.input.placeholder = text

              expect(getCCFieldSubtype(elements.input)).toBe('expiration')
              expect(getUnifiedExpiryDate(elements.input, 12, 2025)).toBe(expectedResult)
            })

            it("matches for label text", () => {
              elements.label.textContent = text

              expect(getCCFieldSubtype(elements.input)).toBe('expiration')
              expect(getUnifiedExpiryDate(elements.input, 12, 2025)).toBe(expectedResult)
            })
        })
    })
```

Reply:
Thanks for the pointer to `describe.each`. I used it here as you describe, but avoided in the other test loop because it was getting more unwieldy (it's not as flexible as a normal loop, especially with naming the test), so I decided to punt. Now that I know about it, I might use it in future tests. Thanks!